diff --git a/build/client/socksclient.js b/build/client/socksclient.js
index 09b1f55767aa7221dee5b78077b2a2bdd332b9c6..23bf6ffc1c5ac8bcc9c6a8a92c8b0cecea24c64c 100644
--- a/build/client/socksclient.js
+++ b/build/client/socksclient.js
@@ -235,7 +235,8 @@ class SocksClient extends events_1.EventEmitter {
             this.socket = existingSocket;
         }
         else {
-            this.socket = new net.Socket();
+            // AbortSignal must be supplied to the Socket constructor; connect() ignores it.
+            this.socket = new net.Socket(this.options.socket_options);
         }
         // Attach Socket error handlers.
         this.socket.once('close', this.onClose);
